File: 5086.js
const input = require('fs')
    .readFileSync('/dev/stdin')
    .toString()
    .trim()
    .split(/\s|\n/)
    .map(Number);

while (input[0] !== 0 && input[1] !== 0) {
    const a = input.shift();
    const b = input.shift();
    
    if (a <= b && b % a === 0) {
        console.log('factor');
    } else if (a >= b && a % b === 0) {
        console.log('multiple');
    } else {
        console.log('neither');
    }
}

File: cpp/5086.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    int A, B;
    cin >> A >> B;

    while (A != 0 && B != 0) {
        if (B % A == 0) {
            cout << "factor" << endl;
        } else if (A % B == 0) {
            cout << "multiple" << endl;
        } else {
            cout << "neither" << endl;
        }

        cin >> A >> B;
    }

    return 0;
}